docs/polygon_draw.js

```diff
@@ -51,11 +50,31 @@ class StoredShapeData {
     this.state = StoredShapeData.State.SAVED;
   }
 
-  /** Decrements write count and pops up 'Saved' message. */
-  noteWriteFinished() {
-    // Keep in sync with code in terminateWriteWithError.
+  /**
+   * Decrements write count and pops up 'Saved' message if all writes done.
+   * @param {boolean} success True if this write succeeded
+   */
+  noteWriteFinished(success) {
     StoredShapeData.pendingWriteCount--;
-    showToastMessage('Saved');
+    if (success) {
+      if (this.errorSaving) {
+        this.errorSaving = false;
+        StoredShapeData.errorCount--;
+      }
+    } else {
+      if (!this.errorSaving) {
+        this.errorSaving = true;
+        StoredShapeData.errorCount++;
+      }
+    }
+    if (StoredShapeData.errorCount === 0 &&
+        StoredShapeData.pendingWriteCount === 0) {
+      showSavedToast();
+    } else if (StoredShapeData.pendingWriteCount === 0 && success) {
+      showToastMessage(
+          'Latest save succeeded, but there are still ' +
+          StoredShapeData.errorCount + ' feature(s) not saved');
```

Comment:
At this point, are these every going to be saved? what does recovery look like?

Reply:
The tests give examples. If you encounter a transient Firestore or EE error, modifying the shape and saving could work: if that happens, we drop the error count above, so we'll know that future saves can be displayed as just having succeeded.

Reply:
Ah helpful - thanks